hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/blockmanagement/TestBlockStatsMXBean.java

```diff
@@ -145,9 +150,11 @@ public void testStorageTypeStatsJMX() throws Exception {
       Map<String,Object> storageTypeStats = (Map<String,Object>)entry.get("value");
       typesPresent.add(storageType);
       if (storageType.equals("ARCHIVE") || storageType.equals("DISK") ) {
-        assertEquals(3l, storageTypeStats.get("nodesInService"));
+        assertEquals(3L, storageTypeStats.get("nodesInService"));
```

Comment:
As this `if-else if-else if-else` getting longer, let's use switch case?

Reply:
`storageType` is a parameter of "java.lang.String" , and `switch()`  does not support "java.lang.String" before java 1.7. So, will `if-else ` be more appropriate here?

Reply:
I have not used Java 7 for a while, but I remember vaguely this is actually supported?

https://docs.oracle.com/javase/specs/jls/se7/html/jls-14.html

Reply:
Hadoop releases before 2.10 are all end of life (EoL). Hadoop 2.10 is the only version using Java 7. We do not need any support, compile or runtime, for Java versions before Java 7.

Hadoop 3.x are all using Java 8+. We do not need any Java 7 support in Hadoop 3.